### PR TITLE
[10.0][MIG]migrate web_widget_timepicker

### PR DESCRIPTION
**Pull request by smurf-U**

### CODE REVIEW

**Comment by Jerther** [comment]
I'm writing a new 10.0 module and I would gladly use this. I'm willing to contribute too if this PR merge can be sped up.

**Comment by pedrobaeza** [comment]
Please review if the migration is correct and put it here.

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 1]
```diff
@@ -3,49 +3,49 @@ odoo.define('web_widget_timepicker', function (require) {
...
-                this.setupFocus(this.$('input'));                   
-            }   
+        initialize_content: function () {
+            if (!this.get("effective_readonly")) {
+                this.$el.find('input').timepicker(this.options);
+                this.setupFocus(this.$('input'));
+            }
         },
-        is_syntax_valid: function() {
+        is_syntax_valid: function () {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and () (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 2]
```diff
@@ -3,49 +3,49 @@ odoo.define('web_widget_timepicker', function (require) {
...
             if (!this.get("effective_readonly") && this.$("input").size() > 0) {
                 try {
-                    this.parse_value(this.$('input').val(),'');
+                    this.parse_value(this.$('input').val(), '');
                     return true;
-                } catch(e) {
+                } catch (e) {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between catch and (e) (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 3]
```diff
@@ -3,49 +3,49 @@ odoo.define('web_widget_timepicker', function (require) {
...
                     return false;
                 }
             }
             return true;
         },
-        is_false: function() {
+        is_false: function () {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and () (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 4]
```diff
@@ -3,49 +3,49 @@ odoo.define('web_widget_timepicker', function (require) {
...
             return this.get('value') === '' || this._super();
         },
-        focus: function() {
+        focus: function () {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and () (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 5]
```diff
@@ -55,35 +55,35 @@ odoo.define('web_widget_timepicker', function (require) {
...
             if (!this.get('effective_readonly')) {
                 this.internal_set_value(
                     this.parse_value(
-                        this.$('input').val(),''));
+                        this.$('input').val(), ''));
             }
         },
-        parse_value: function(val, def) {
-            return formats.parse_value(val, {"widget": this.internal_format}, def);  
+        parse_value: function (val, def) {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and (val, def) (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 6]
```diff
@@ -55,35 +55,35 @@ odoo.define('web_widget_timepicker', function (require) {
...
+            return formats.parse_value(val, {"widget": this.internal_format}, def);
         },
-        format_value: function(val, def) {
+        format_value: function (val, def) {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and (val, def) (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 7]
```diff
@@ -55,35 +55,35 @@ odoo.define('web_widget_timepicker', function (require) {
...
             return formats.format_value(val, {"widget": this.internal_format}, def);
         },
-        render_value: function() {
-            var show_value = this.format_value(this.get('value'),'');             
+        render_value: function () {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between function and () (preserve coding style accross file)

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 8]
```diff
@@ -3,49 +3,49 @@ odoo.define('web_widget_timepicker', function (require) {
...
                 scrollDefault: 'now',
             });
         },
-        initialize_content: function() {
-            if(!this.get("effective_readonly")) {
-                this.$el.find('input').timepicker(this.options);                    
-                this.setupFocus(this.$('input'));                   
-            }   
+        initialize_content: function () {
+            if (!this.get("effective_readonly")) {
```
Comment (Jerther) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Remove space between if and () (preserve coding style accross file)

File: web_widget_timepicker/static/src/xml/web_widget_timepicker.xml [hunk 9]
```diff
@@ -14,7 +14,7 @@
                 <span class="fa fa-clock-o o_timepicker_button"/>
```
Comment (Jerther) [inline web_widget_timepicker/static/src/xml/web_widget_timepicker.xml:14]: Clock icon appears under dropdown box
image

**Review by Jerther — CHANGES_REQUESTED** [review]
README.rst needs to be replaced with new template. See  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

**Review by Jerther — CHANGES_REQUESTED** [review]
It would also be a good opportunity to upgrade the jquery plugin which is v1.10.1 to v1.11.11

**Comment by smurf-U** [comment]
@Jerther jquery plugin v1.10.1 upgrade in last commit to v1.11.11 and done with all changs..

**Review by Jerther — APPROVED** [review]
That was quick! :)
Looks good to me!

**Comment by moylop260** [comment]
Could you follow the script:
 - https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0

**Comment by moylop260** [comment]
I'm using the runbot instance and I just change the Stock's view for the fields.Datetime `min_date` to add this widget.
 - screen shot 2017-07-21 at 14 27 40
   - http://3298939-656-b97d2f.runbot2.odoo-community.org/web?debug=1#id=698&view_type=form&model=ir.ui.view&action=25

But is not displayed:
 - screen shot 2017-07-21 at 14 27 56
   - http://3298939-656-b97d2f.runbot2.odoo-community.org/web?debug=1#id=7&view_type=form&model=stock.picking&action=301&active_id=1

What is wrong?

**Comment by smurf-U** [comment]
@moylop260 why you use timepicker widget with field whose type datetime? odoo have default datetime widget.

**Comment by Jerther** [comment]
@moylop260  The underlying field must be `fields.Float()`.

**Comment by moylop260** [comment]
Thank for clarify
 - screen shot 2017-07-21 at 15 26 53

👍

**Comment by Jerther** [comment]
Ping

**Comment by katyukha** [comment]
Hope this PR will be merged soon!

Thanks for porting)

**Comment by katyukha** [comment]
@moylop260 thanks)

**Comment by vcloud9ray** [comment]
In the Enterprise Edition, it shows the times perfectly in the dropdown, but if you click 7.30 then the value in the field turns into 7.5.

Did the value after selection match the value in the dropdown when you tested?

If you TYPE the value 7:30 then 7:30 is shown.  Weird.

**Comment by vcloud9ray** [comment]
Had to change the python field to Char and also change the JavaScript internal_format: 'char' for this to work properly (show 8:30 in the field instead of 8.5 when 8:30 was selected).  Then, behind the scenes I needed to create a float version of the field and deduce the float_time value.

**Comment by vcloud9ray** [comment]
Had to comment out the `<span class="fa fa-clock-o o_timepicker_button"/>` because in Enterprise is doesn't sit beside the field, but rather shows up in the middle of other fields in the list and looks very ugly.

**Pull request merged**